extensions/interactions/PencilCodeEditor/PencilCodeEditor.js

```diff
@@ -69,7 +69,25 @@ oppia.directive('oppiaInteractivePencilCodeEditor', [
           });
 
           $scope.reset = function() {
-            pce.setCode($scope.initialCode);
+            $modal.open({
+              templateUrl: 'modals/pencilCodeResetConfirmation',
+              backdrop: 'static',
+              keyboard: false,
+              controller: [
+                '$scope', '$modalInstance', function($scope, $modalInstance) {
+                  $scope.cancel = function() {
+                    $modalInstance.dismiss();
+                  };
+
+                  $scope.resetCode = function() {
+                    $modalInstance.close(true);
```

Comment:
yes, I can do away with it by shifting the resetting thing inside this function itself. Should I do that ?